lib/daru/dataframe.rb

```diff
@@ -357,13 +356,7 @@ def [](*names)
         axis = :vector
       end
 
-      if axis == :vector
-        access_vector(*names)
-      elsif axis == :row
-        access_row(*names)
-      else
-        raise IndexError, "Expected axis to be row or vector not #{axis}"
-      end
+      dispatch_to_axis axis, :access, *names
```

Comment:
I promised not to do a refactorings in this PR... But there are just too many similar cases to cover (check axis, call appropriate method, throw if axis is wrong), so ....